### PR TITLE
Defensively prevented double trailing slash for IDPs with trailing slashes

### PR DESCRIPTION
**Pull request by jaxoncreed**

### CODE REVIEW

File: src/ProvidersCache.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ class ProvidersCache {
    * @returns {Promise}
    */
   discover (issuer) {
-    return fetch(`${issuer}/.well-known/openid-configuration`)
+    return fetch(`${issuer}${issuer.endsWith('/') ? '' : '/'}.well-known/openid-configuration`)
```
Comment (RubenVerborgh) [inline src/ProvidersCache.js]: Let's just catch all cases with:
```suggestion
    return fetch(issuer.replace(/\/*$/, '/.well-known/openid-configuration')
```

**Comment by jaxoncreed** [comment]
Apparently there were some failing tests from before, so I fixed those too.

**Comment by timbl** [comment]
This suggests the apis are sloppy about whether an origin is represented with or without a trailing space. If now either is allowed in turn documented? Has someone thought this through or just slapped on a quick fix?

**Review by michielbdejong — APPROVED** [review]
I agree with Tim, can you find out who is writing those trailing slashes into the data, and fix the problem at the source?

At the same time, while that is being fixed, I think it's in the interest of the user for NSS to be forgiving, so we can merge this as a work-around. Maybe add a code comment here that points to the bug at the source?

**Comment by michielbdejong** [comment]
@jaxoncreed ping me after you merged this, I'll use https://github.com/solid/solidos/blob/master/scripts/update-tree.sh to update the entire dependency tree of NSS, including this patch.

**Comment by jaxoncreed** [comment]
@timbl @michielbdejong Trailing slashes are allowed in the OIDC spec https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest, and must be accounted for.

**Pull request merged**